doc/api/customization.md

```diff
@@ -0,0 +1,51 @@
+# Customization
+
+> Stability: 1.0 - Early development
+
+Node.js has multiple APIs for customizing behaviour and can easily be invoked
+via [`--import`][]: A `"nodejs-customization"` [entry point][`"exports"`] for a
+package loaded via [`--import`][] is automatically run at startup. The
+package.json of such a plugin would look something like this:
+
+```json
+{
+  "name": "example-nodejs-plugin",
+  "keywords": [
+    "nodejs-plugin"
+  ],
+  "exports": {
+    "nodejs-customization": "./registration.mjs"
+  }
+}
+```
+
+Setting the keyword `nodejs-plugin` is important for users to find the package
+(which may be automated). It should also contain the official name(s) for the
+support it provides; for a plugin extending support for typescript would contain
+`typescript` in its keywords; avoid red-herrings such as `typescript` as a
+keyword when the library does not extend support for typescript but is merely
+itself written in typescript. Failing to avoid these red-herrings will cause
+your package to be listed for things it does not do, aggravating users.
+
+These steps configure Node.js with a customization plugin:
+1. Install the plugin via your preferred package manager. For finding the
+   plugin, package managers provide often a CLI utility, such as [`npm
+   search`][], as well as a website.
+1. Once installed, in order to get Node.js to automatically use it, create a
+   [`.env`][`--env-file`] file containing an [`--import`][] for your chosen
+   plugin, like `NODE_OPTIONS="--import=example-nodejs-plugin"`.
```

Comment:
If we’re using the standardized export that you mentioned above, this would be

```suggestion
   plugin, like `NODE_OPTIONS="--import=example-nodejs-plugin/nodejs-customization"`.
```

Reply:
OH!! Actually, this is an interesting point, and I think absolves us from a heinous entry-point name: We "control" this `--import`, so if the `--import` itself specifies the entry-point, I think that means it's safe to use whatever we want (so this would be okay):

```json
{
  "name": "example-nodejs-plugin",
  "exports": {
    "register": "whatever.mjs"
  },
  "keywords": ["nodejs-plugin", …]
}
```

```shell
NODE_OPTIONS="--import=example-nodejs-plugin/register"
```

Reply:
Right, but that’s also why it’s not that important that it be standardized. I think the only benefit of standardization is so that the wizard we eventually create knows what export to use.

Reply:
> […] the wizard we eventually create knows what export to use.

Yes, exactly.